src/mobile/components/app_banner/test/app_banner.test.coffee

```diff
@@ -13,6 +13,8 @@ describe 'AppBanner', ->
     benv.teardown()
 
   beforeEach ->
+    window.location = {}
+    global.location = window.location
```

Comment:
It's probably fine, but these should be assigned back to the original object in an `afterEach` or `after`.

Reply:
Good call, I think part of why the mocha tests are so fragile is because overwritten globals are not getting torn down.

Reply:
Changed this to not assign the empty object, but unsure how to tear it down.  Going to merge this since it resolves a handful of sticky tests introduced in an earlier PR, but would be glad to pair on a better solution Monday.